catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -166,33 +180,34 @@ public void createContentItem(FileSystemPersistenceProvider fileIdMap, ContentEn
             if (createResponse != null) {
                 List<Metacard> createdMetacards = createResponse.getCreatedMetacards();
 
-                for (Metacard metacard : createdMetacards) {
-                    fileIdMap.store(key, metacard.getId());
-                    LOGGER.debug("content item created with id = {}", metacard.getId());
+                if (safeKey != null) {
+                    fileIdMap.store(safeKey,
+                            createdMetacards.get(0)
```

Comment:
❓ Why just the first metacard in the list? Should we protect against null or empty `createdMetacards`?